benchmark/test/test_httpapi.py

```diff
@@ -272,3 +276,139 @@ def test_delete_nonexistent(self):
         req = self.agent.request("DELETE", location)
         req.addCallback(self.check_response_code, http.NOT_FOUND)
         return req
+
+    BRANCH1_RESULT1 = {u"userdata": {u"branch": u"1"}, u"value": 100,
+                       u"timestamp": datetime(2016, 1, 1, 0, 0, 5).isoformat()}
+    BRANCH1_RESULT2 = {u"userdata": {u"branch": u"1"}, u"value": 120,
+                       u"timestamp": datetime(2016, 1, 1, 0, 0, 6).isoformat()}
+    BRANCH2_RESULT1 = {u"userdata": {u"branch": u"2"}, u"value": 110,
+                       u"timestamp": datetime(2016, 1, 1, 0, 0, 7).isoformat()}
+
+    def setup_results(self):
+        """
+        Submit some results for testing various queries against them.
+        """
+        results = [
+            self.BRANCH1_RESULT1, self.BRANCH1_RESULT2, self.BRANCH2_RESULT1
+        ]
+        shuffle(results)
```

Comment:
This will make the tests non-deterministic. If this change is to prove that sorting works, it would be better to insert them out of order.